cpp/ql/lib/semmle/code/cpp/models/implementations/StdContainer.qll

```diff
@@ -118,6 +118,8 @@ private class StdSequenceContainerData extends TaintFunction {
     input.isReturnValueDeref() and
     output.isQualifierObject()
   }
+
+  override predicate isPartialWrite(FunctionOutput output) { output.isQualifierObject() }
```

Comment:
Yeah, a partial write is probably the norm. There are obvious outliers such as [std::string::clear](https://en.cppreference.com/w/cpp/string/basic_string/clear), but I would also expect it to be the norm.

I'm not sure how to structure this best, though. Since we're now "blocking by default" and have thus added a predicate to "disable blocking" there's no now predicate to override for "enabling blocking". If we should make writes to the qualifier non-blocking by default we should also have a "enabling blocking" predicate. I supposed that could simply be a matter of adding an overridable predicate (with a sensible default) to `PartialFlowFunction` that says that writes to qualifiers are always non-blocking.

But I'd like to leave that as a follow-up like you suggested.